.github/workflows/markdownlint.yml

```diff
@@ -9,13 +9,17 @@ on:
 jobs:
   run-markdownlint:
     runs-on: ubuntu-22.04
+    permissions:
+      contents: read
 
     steps:
     - name: check out code
       uses: actions/checkout@v4
+      with:
+        persist-credentials: false
 
     - name: run markdownlint
-      uses: DavidAnson/markdownlint-cli2-action@v18.0.0
+      uses: DavidAnson/markdownlint-cli2-action@992badcdf24e3b8eb7e87ff9287fe931bcb00c6e
```

Comment:
curious - why not use a stable version but a hash?

Reply:
zizmore recommends this, because the stable version is something that the author of the 3p action can change - whereas the hash cant be changed